File: py/p6.py
def square_of_sum(n: int) -> int:
    sum = n * (n+1) / 2
    return int(pow(sum, 2))

def sum_of_squares(n: int) -> int:
    return int(n * (n+1) * (2*n+1) / 6)

def diff(n: int) -> int:
    return square_of_sum(n) - sum_of_squares(n)

print(diff(100))

File: rs/p6.rs
fn square_of_sum(n: u128) -> u128 {
    let sum = n * (n + 1) / 2;
    sum.pow(2)
}

fn sum_of_squares(n: u128) -> u128 {
    n * (n + 1) * (2 * n + 1) / 6
}

fn diff(n: u128) -> u128 {
    square_of_sum(n) - sum_of_squares(n)
}

fn main() {
    println!("{}", diff(100));
}
